agent.py

```diff
@@ -196,6 +206,266 @@ def _do_restart(self):
+            'hostname': self._hostname,
+            'email': email
+        }
+        r = requests.post(url, files=files, data=data)
+        self._analyse_result(r)
+
+    # Start by creating the tar file which will contain everything
+    def _init_tarfile(self):
+        # Default temp path
+        self._tar_path = os.path.join(tempfile.gettempdir(), 'datadog-agent.tar.bz2')
+
+        if os.path.exists(self._tar_path):
+            os.remove(self._tar_path)
+        self._tar = tarfile.open(self._tar_path, 'w:bz2')
+
+    # Save logs file paths
+    def _save_logs_path(self, config):
+        prefix = ''
+        if get_os() == 'windows':
+            prefix = 'windows_'
+        self._collector_log = config.get('{0}collector_log_file'.format(prefix))
+        self._forwarder_log = config.get('{0}forwarder_log_file'.format(prefix))
+        self._dogstatsd_log = config.get('{0}dogstatsd_log_file'.format(prefix))
+        self._jmxfetch_log = config.get('jmxfetch_log_file')
+
+    # Add logs to the tarfile
+    def _add_logs_tar(self):
+        self._add_log_file_tar(self._collector_log)
+        self._add_log_file_tar(self._forwarder_log)
+        self._add_log_file_tar(self._dogstatsd_log)
+        self._add_log_file_tar(self._jmxfetch_log)
+        self._add_log_file_tar(
+            "{0}/*supervisord.log*".format(os.path.dirname(self._collector_log))
+        )
+
+    def _add_log_file_tar(self, file_path):
+        for f in glob.glob('{0}*'.format(file_path)):
+            self._print("  * {0}".format(f))
+            self._tar.add(
+                f,
+                os.path.join(self._prefix, 'log', os.path.basename(f))
+            )
+
+    # Collect all conf
+    def _add_conf_tar(self):
+        conf_path = get_config_path()
+        self._print("  * {0}".format(conf_path))
+        self._tar.add(
+            self._strip_comment(conf_path),
+            os.path.join(self._prefix, 'etc', 'datadog.conf')
+        )
+
+        if get_os() != 'windows':
+            supervisor_path = os.path.join(
+                os.path.dirname(get_config_path()),
+                'supervisor.conf'
+            )
+            self._print("  * {0}".format(supervisor_path))
+            self._tar.add(
+                self._strip_comment(supervisor_path),
+                os.path.join(self._prefix, 'etc', 'supervisor.conf')
+            )
+
+        for file_path in glob.glob(os.path.join(get_confd_path(), '*.yaml')):
+            self._add_clean_confd(file_path)
+
+    # Return path to a temp file without comment
+    def _strip_comment(self, file_path):
+        _, temp_path = tempfile.mkstemp(prefix='dd')
+        atexit.register(os.remove, temp_path)
+        temp_file = open(temp_path, 'w')
+        orig_file = open(file_path, 'r').read()
+
+        for line in orig_file.splitlines(True):
+            if not self.COMMENT_REGEX.match(line) and not self.APIKEY_REGEX.match(line):
+                temp_file.write(line)
```

Comment:
We can probably redact only part of the API key, `XXXXXXXXXXXXXXXXXXXXXXbcdef` so we can still have a good chance of finding it in the DB without compromising it?